File: examples/javascript/basic-research.js
/**
 * Basic Research Example
 * Demonstrates how to use NSAI Data for simple research queries
 */

import { NSAIClient } from 'nsai-sdk';

// Initialize the client
const client = new NSAIClient(process.env.NSAI_API_KEY);

async function main() {
    try {
        // Create a research query
        console.log('🔍 Starting research on AI ethics...');
        
        const response = await client.research(
            'What are the main ethical considerations in AI development?',
            {
                maxSources: 15,
                outputFormat: 'markdown',
                enableValidation: true
            }
        );
        
        // Display results
        console.log(`\n✅ Research completed in ${response.durationMs}ms`);
        console.log(`📊 Analyzed ${response.sources?.length || 0} sources`);
        
        if (response.metadata?.validation?.confidence_score) {
            console.log(`🎯 Confidence score: ${response.metadata.validation.confidence_score}%`);
        }
        
        console.log('\n📄 Report:');
        console.log('-'.repeat(80));
        console.log(response.report);
        console.log('-'.repeat(80));
        
        // Save report to file
        const fs = require('fs').promises;
        await fs.writeFile('ai_ethics_research.md', response.report);
        console.log('\n💾 Report saved to ai_ethics_research.md');
        
    } catch (error) {
        console.error('❌ Error:', error.message);
    }
}

main();

File: examples/python/basic_research.py
"""
Basic Research Example
Demonstrates how to use NSAI Data for simple research queries
"""

from nsai import NSAIClient
import os

# Initialize the client
# You can pass the API key directly or set NSAI_API_KEY environment variable
client = NSAIClient(api_key=os.getenv("NSAI_API_KEY"))

def main():
    # Create a research query
    print("🔍 Starting research on quantum computing...")
    
    response = client.research(
        query="What are the latest breakthroughs in quantum computing in 2024?",
        max_sources=15,
        output_format="markdown"
    )
    
    # Display results
    print(f"\n✅ Research completed in {response.duration_ms}ms")
    print(f"📊 Analyzed {len(response.sources)} sources")
    
    if response.confidence_score:
        print(f"🎯 Confidence score: {response.confidence_score}%")
    
    print("\n📄 Report:")
    print("-" * 80)
    print(response.report)
    print("-" * 80)
    
    # Save report to file
    with open("quantum_computing_research.md", "w") as f:
        f.write(response.report)
    print("\n💾 Report saved to quantum_computing_research.md")

if __name__ == "__main__":
    main()